lib/Bridge/TolerantParser/ChainTolerantCompletor.php

```diff
@@ -53,9 +55,16 @@ public function complete(TextDocument $source, ByteOffset $byteOffset): Generato
                 continue;
             }
 
+            $importTable = $this->getClassImportTablesForNode($completionNode);
             $suggestions = $tolerantCompletor->complete($completionNode, $source, $byteOffset);
 
-            yield from $suggestions;
+            foreach ($suggestions as $suggestion) {
+                // Trick to avoid any BC break when converting to an array
+                // https://www.php.net/manual/fr/language.generators.syntax.php#control-structures.yield.from
+                foreach ($this->resolveAliasSuggestions($importTable, $suggestion) as $resolvedSuggestion) {
+                    yield $resolvedSuggestion;
+                }
+            }
```

Comment:
could this be a new decorator rather than modifying this class?

Reply:
I like the idea but I'm not sure it would allow us to deal with every possible case.
For example the `DoctrineAnnotationCompletor` also needs to update the `snippet` of each suggestion.

I'll think about it and gave it a try, if there is a way to update the snippet in the same way for all classes then yes and it will solve your next review too.